packages/sources/view-function/README.md

```diff
@@ -0,0 +1,69 @@
+# Chainlink External Adapter for View-Function
+
+External adapter for executing contract function and returning the result
+
+### Environment Variables
+
+The adapter takes the following environment variables:
+
+| Required? |  Name   |     Description     | Options | Defaults to |
+| :-------: | :-----: | :-----------------: | :-----: | :---------: |
+|    ✅     | RPC_URL | RPC URL of ETH node |         |             |
+
+---
+
+### Input Parameters
+
+| Required? |   Name   |     Description     |            Options             | Defaults to |
+| :-------: | :------: | :-----------------: | :----------------------------: | :---------: |
+|           | endpoint | The endpoint to use | [function](#Function-Endpoint) |  function   |
+
+---
+
+## Function Endpoint
+
+### Input Params
+
+| Required? |           Name            |                                                                         Description                                                                         | Options | Defaults to |
+| :-------: | :-----------------------: | :---------------------------------------------------------------------------------------------------------------------------------------------------------: | :-----: | :---------: |
+|    ✅     |  `address` or `contract`  |                                                                   Address of the contract                                                                   |         |             |
+|    ✅     | `function` or `signature` | Function signature. Should be formatted as [human readable ABI](https://docs.ethers.io/v5/single-page/#/v5/getting-started/-%23-getting-started--contracts) |         |             |
+|           |       `inputParams`       |                                                            Array of function parameters in order                                                            |         |             |
+
+### Sample Input
+
+```json
+{
+  "id": 1,
+  "data": {
+    "contract": "0xdAC17F958D2ee523a2206206994597C13D831ec7",
+    "function": "function symbol() view returns (string)"
+  }
+}
+```
+
+### Sample Output
+
+```json
+{
+  "jobRunID": 1,
+  "result": "0x0000000000000000000000000000000000000000000000000000000055534454",
+  "statusCode": 200,
+  "data": {
+    "result": "0x0000000000000000000000000000000000000000000000000000000055534454"
```

Comment:
The encoded result of a string should be larger than this. For example:
```
0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000045553445400000000000000000000000000000000000000000000000000000000
```
We should be replicating the encoding for any type rather than converting to hex.